### DISCUSSION TITLE
djot+pandoc

### DISCUSSION THREAD

**Discussion by olynch** [jgm/djot#222] [General]
Is there a way to use djot with pandoc? I have a hakyll based blog, and I'd like to write posts in djot. Also, I have a bunch of sites built on quarto, which uses pandoc under the hood, and it would be cool to have djot support there.

**Comment by jgm** [comment]
Pandoc doesn't have djot support built in, but the lua implementation of djot includes a custom reader and writer that can be used with pandoc, and the js implementation of djot can convert between djot and pandoc's AST (in JSON format).  In principle it would be possible to use these with hakyll.  Using the (better supported) djot.js would require shelling out, but with the custom reader and writer you could use the pandoc + pandoc-lua-engine libraries.

**Comment by the-solipsist** [comment]
Is there a reason that pandoc isn't yet incorporating djot support as a built-in feature? Is it a matter of time and energy (which would be just fine), or do you (@jgm) wish to see djot mature a bit more before incorporating it into pandoc, or is it something else?  If pandoc incorporates djot, then other projects that use pandoc (like quarto) could do so as well.

**Comment by jgm** [comment]
A bit of both I guess.  But Hécate is working on a Haskell djot parser.  When that is done it should be easy to add built-in pandoc support, and I probably will: https://github.com/Kleidukos/djot-hs
